source/AI.cpp

```diff
@@ -2856,6 +2856,23 @@ Point AI::TargetAim(const Ship &ship, const Body &target)
 
 
 
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint)
+{
+	shared_ptr<const Body> target = ship.GetTargetShip() ?: ship.GetTargetAsteroid();
```

Comment:
```suggestion
	shared_ptr<const Body> target = ship.GetTargetShip() ? ship.GetTargetShip() : ship.GetTargetAsteroid();
```

Reply:
The current code is short hand for the suggestion.

https://gcc.gnu.org/onlinedocs/gcc-5.3.0/gcc/Conditionals.html#Conditionals
https://stackoverflow.com/questions/1785426/c-sharp-null-coalescing-operator-equivalent-for-c
http://www.dev-tricks.net/the-operator-aka-null-coalescing-operator/

Reply:
Apparently the ?: operator, both versions, require the same type. So changed thing up.